pypesto/objective/amici/amici.py

```diff
@@ -456,7 +456,7 @@ def call_unprocessed(
 
         nllh = ret[FVAL]
         rdatas = ret[RDATAS]
-        if INNER_PARAMETERS in ret and ret[INNER_PARAMETERS]:
+        if INNER_PARAMETERS in ret and any(ret[INNER_PARAMETERS]):
             self.inner_parameters = ret[INNER_PARAMETERS]
```

Comment:
Hmmm this was a mistake. It should be `ret[INNER_PARMETERS] is not None`. It can be `None` if the simulation fails, or if the `inner_calculator_collector` doesn't have any interpretable inner parameters (basically noise, spline or ordinal parameters are not easily interpretable by just number).

Reply:
Ah OK, then can be `if ret.get(INNER_PARAMETERS, None) is not None:`

Reply:
Ohh that's even nicer :D 
Will make it into that.